tests/components/laundrify/test_coordinator.py

```diff
@@ -1,52 +1,59 @@
 """Test the laundrify coordinator."""
 
+from datetime import timedelta
+
+from freezegun.api import FrozenDateTimeFactory
 from laundrify_aio import exceptions
 
-from homeassistant.components.laundrify.const import DOMAIN
+from homeassistant.components.laundrify.const import DEFAULT_POLL_INTERVAL, DOMAIN
 from homeassistant.core import HomeAssistant
 
-from . import create_entry
+from tests.common import async_fire_time_changed
 
 
-async def test_coordinator_update_success(hass: HomeAssistant) -> None:
+async def test_coordinator_update_success(
+    hass: HomeAssistant,
+    laundrify_config_entry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
     """Test the coordinator update is performed successfully."""
-    config_entry = create_entry(hass)
-    await hass.config_entries.async_setup(config_entry.entry_id)
-    await hass.async_block_till_done()
-    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
-    await coordinator.async_refresh()
+    freezer.tick(timedelta(seconds=DEFAULT_POLL_INTERVAL))
+    async_fire_time_changed(hass)
     await hass.async_block_till_done()
 
+    coordinator = hass.data[DOMAIN][laundrify_config_entry.entry_id]["coordinator"]
     assert coordinator.last_update_success
```

Comment:
So instead of asserting this, we want to assert that our entities are unavailable (that is the default behaviour for `CoodinatorEntity`s). This way we don't touch internals

Reply:
Haha, I think I'm slowly getting where you want me to be 😄

Reply:
Refactored in 87047a13cf3f2b34adb6b08da5fc936b5d5b69ed